deepchecks/nlp/utils/text_properties.py

```diff
@@ -295,265 +365,223 @@ def _predict(text, classifier, kind):
     )
-    return [
-        _predict(text, classifier, 'toxicity')
-        for text in raw_text
-    ]
+    return _predict(text, classifier, 'toxicity')
 
 
 def fluency(
-    raw_text: Sequence[str],
-    device: Optional[int] = None,
-    models_storage: Union[pathlib.Path, str, None] = None
-) -> List[float]:
-    """Return list of floats of fluency."""
+        text: str,
+        device: Optional[int] = None,
+        models_storage: Union[pathlib.Path, str, None] = None
+) -> float:
+    """Return float representing fluency."""
     model_name = 'prithivida/parrot_fluency_model'
     classifier = get_transformer_pipeline(
         'fluency',
         model_name,
         device=device,
         models_storage=models_storage
     )
-    return [
-        _predict(text, classifier, 'fluency')
-        for text in raw_text
-    ]
+    return _predict(text, classifier, 'fluency')
 
 
 def formality(
-    raw_text: Sequence[str],
-    device: Optional[int] = None,
-    models_storage: Union[pathlib.Path, str, None] = None
-) -> List[float]:
-    """Return list of floats of formality."""
+        text: str,
+        device: Optional[int] = None,
+        models_storage: Union[pathlib.Path, str, None] = None
+) -> float:
+    """Return float representing formality."""
     model_name = 's-nlp/roberta-base-formality-ranker'
     classifier = get_transformer_pipeline(
         'formality',
         model_name,
         device=device,
         models_storage=models_storage
     )
-    return [
-        _predict(text, classifier, 'formality')
-        for text in raw_text
-    ]
+    return _predict(text, classifier, 'formality')
 
 
-def lexical_density(raw_text: Sequence[str]) -> List[str]:
-    """Return a list of floats of lexical density per text sample.
+def lexical_density(text: str) -> float:
+    """Return a float representing lexical density.
 
     Lexical density is the percentage of unique words in a given text. For more
     information: https://en.wikipedia.org/wiki/Lexical_density
     """
+    if pd.isna(text):
+        return np.nan
     if not nltk_download('punkt', quiet=True):
-        warnings.warn('nltk punkt not found, lexical density cannot be calculated.'
-                      ' Please check your internet connection.', UserWarning)
-        return [np.nan] * len(raw_text)
-    result = []
-    for text in raw_text:
-        if not pd.isna(text):
-            all_words = textblob.TextBlob(text).words
-            if len(all_words) == 0:
-                result.append(np.nan)
-            else:
-                total_unique_words = len(set(all_words))
-                text_lexical_density = round(total_unique_words * 100 / len(all_words), 2)
-                result.append(text_lexical_density)
-        else:
-            result.append(np.nan)
-    return result
+        _warn_if_missing_nltk_dependencies('punkt', 'Lexical Density')
+        return np.nan
+
+    all_words = _word_tokenize_with_cache(text)
```

Comment:
Does not work for lexical density, tried it. Because it measures the ratio of unique to non-unique, sampling favors non-unique words and it affects the result significantly.